splink/comparison_template_library.py

```diff
@@ -1170,6 +1174,9 @@ def __init__(
 
         comparison_levels = []
 
+        if set_to_uppercase:
+            postcode_col = f"upper({col_name})"
```

Comment:
I think this should be using `postcode_col.sql()` as the column name.

If we encounter an invalid column name (a name with spaces, a name such as "group", etc), this code breaks, I believe.

Reply:
Yeah, just to confirm, this doesn't appear to work.

You'd need to put `postcode_col` inside all of the existing `_exact_match_level` calls.

Reply:
Sorry, I think this was wrong. Have changed to `col_name = f"upper({col_name})"` to ensure the capitalisation pulls through the rest of the comparison levels

Reply:
That also works.